source/simulator/nullspace.cc

```diff
@@ -385,33 +385,13 @@ namespace aspect
           const std::vector<Point<dim> > &q_points = fe.get_quadrature_points();
 
           // get the density at each quadrature point if necessary
-          MaterialModel::MaterialModelInputs<dim> in(n_q_points,
-                                                     parameters.n_compositional_fields);
+          MaterialModel::MaterialModelInputs<dim> in(fe,
+                                                     &cell,
+                                                     this->introspection,
+                                                     relevant_dst);
           MaterialModel::MaterialModelOutputs<dim> out(n_q_points,
-                                                       parameters.n_compositional_fields);
-
-          //Get the velocity at each quadrature point
-          fe[introspection.extractors.velocities].get_function_values (relevant_dst, in.velocity);
-
-          if ( ! use_constant_density)
-            {
-              fe[introspection.extractors.pressure].get_function_values (relevant_dst, in.pressure);
-              fe[introspection.extractors.temperature].get_function_values (relevant_dst, in.temperature);
-              fe[introspection.extractors.pressure].get_function_gradients (relevant_dst, in.pressure_gradient);
-              for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
-                fe[introspection.extractors.compositional_fields[c]].get_function_values(relevant_dst,
-                                                                                         composition_values[c]);
+                                                       introspection.n_compositional_fields);
 
-              for (unsigned int i=0; i<n_q_points; ++i)
-                {
-                  in.position[i] = fe.quadrature_point(i);
-                  for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
-                    in.composition[i][c] = composition_values[c][i];
-                }
-              in.cell = &cell;
-
-              material_model->evaluate(in, out);
```

Comment:
this change doesn't look right. you used to call `material_model->evaluate()` before, but now you no longer do.

In fact, this may not be the very best place to test your changes. We really only need anything other than the velocities in `in` if `use_constant_density == false`. You now do substantially more work here by populating all of the fields (and adding back the now missing call to `evaluate`).

Maybe the best way to deal with this would be to leave this one place as it was before (can also easily be done with git) and pick a different place where we initialize a `MaterialModelInputs` object to test your new code.